docs/database-functionalities/streams/pulsar/pulsar-streams.md

```diff
@@ -0,0 +1,141 @@
+[reference guide](/reference-guide/streams/overview.md).
+
+:::
+
+## Configuring Memgraph
+
+You need to provide a service URL so Memgraph can locate the Pulsar cluster. The
+service URL can be set using the configuration flag `--pulsar-service-url` .
+
+## Creating the stream
+
+The very first step is to make sure at least one transformation module is loaded
+into Memgraph. If you are not sure how to define them, check out the
+[transformation module
+guide](/database-functionalities/streams/pulsar/implement-transformation-module.md).
+We will use `transformation.my_pulsar_transformation` from that example, but we
+are going to alias it as `my.pulsar_transform` to make the size of result tables
+slimmer. For the topic name, we are going to use `topic1` .
+
+```cypher
+CREATE PULSAR STREAM myStream
+TOPICS topic1
+TRANSFORM my.pulsar_transform;
+```
+
+Check the created stream:
+
+```cypher
+SHOW STREAMS;
+```
+
+The result should be similar to:
+
+```plaintext
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+--------------------
+| name                  | topics     | batch_interval        | batch_size            | transformation_name          | owner                 | is running            |
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+--------------------
+| "myStream"            | ["topic1"] | Null                  | Null                  | "my.pulsar_transform"        | Null                  | false                 |
++-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+-----------------------+--------------------
+
+```
+
+The result contains the most important information about the existing streams,
+e.g., its name, topics it is subscribed to, etc.
+
+## Check if the stream is working
+
+:::warning
+
+`CHECK STREAM` only works for single non-partitioned topic consumers. The next
+examples will fail if your Pulsar stream source consumes multiple topics or from
+a partitioned topic.
+
+:::
+
+Maybe at first, you don't want to run the stream in the background but see the
+actual result of the transformation. This can be handy when implementing a
+transformation. To achieve that, we can use the `CHECK STREAM` query. This query
+will consume the message from the last acknowledged message but won't
+acknowledge the next message. That means you are free to play around with it,
+and there won't be any permanent effects. For a freshly created stream, there is
+probably no acknowledged message, so the `CHECK STREAM` query will wait for new
+messages. By default, the query will wait `30000` milliseconds ( `30` seconds)
+and after that, it will throw a timeout exception. To give us some more time,
+use a larger timeout, e.g.: `60000` milliseconds ( `60` seconds):
```

Comment:
Is the behaviour for pulsar the same? Does it consume the message?

Reply:
yes it is, no acknowledgment is done.